django_toolkit/fallbacks/circuit_breaker.py

```diff
@@ -34,21 +34,25 @@ def total_failures(self):
         return self.cache.get(self.failure_cache_key) or 0
 
     def open_circuit(self):
+        self.cache.set(self.circuit_cache_key, True, self.circuit_timeout)
+
         logger.critical(
             'Open circuit for {failure_cache_key} {cicuit_cache_key}'.format(
                 failure_cache_key=self.failure_cache_key,
                 cicuit_cache_key=self.circuit_cache_key
             )
         )
 
-        self.cache.set(self.circuit_cache_key, True, self.circuit_timeout)
-
     def __enter__(self):
         if self.is_circuit_open:
             raise self.max_failure_exception
 
+        return self
+
     def __exit__(self, exc_type, exc_value, traceback):
         if exc_type in self.catch_exceptions:
+            if self.is_circuit_open:
+                raise self.max_failure_exception
 
             self._increase_failure_count()
 
```

Comment:
Wouldn't it be better to check `is_circuit_open` in the `if` below (Something like: `not self.is_circuit_open and (self.total_failures >= self.max_failures)`).

I don't know if there's any value in raising an exception after the code inside the context is executed. If it succeeded, we would raise an exception for no reason.